### DISCUSSION TITLE
Changing or adding keys

### DISCUSSION THREAD

**Discussion by furry3000** [opensteno/plover#1492] [General]
What if I was setting up a mechanical keyboard and I wanted extra keys? I would like to do something similar with other keys also. I believe this would help with my Plover speed. The idea is similar to the LightSpeed steno writer: It has a little bump between keys for hitting two keys at once, easier; https://i0.wp.com/www.stenovations.com/wp-content/uploads/2012/02/Zenith-II.jpg?ssl=1. 
But I would be using a mechanical keyboard.

Example:
Key 1 ........ Key 2 ...... Key3 ...... Key4 .... Key5 .... Key6
  **A .............. AO ........... O ...........  E ........... EU ......... U**
 (x) .............. (c) ............ (v) ........... (n) ......... (m) ....... (,) 

If I Plover had some type of graphic that I would use to make a change, that would be easiest. But I'm willing to do programming lines if I had the instructions to do so.

Please Help,
--Furry3000

**Comment by user202729** [comment]
This is possible, but definitely not easy to do. You need a new [system plugin](https://github.com/openstenoproject/plover/wiki/Plugins), and some way to transform the existing dictionary to suit the new system.

Although in this case, I guess making the thumb span 3 keys is not exactly comfortable. You can try using/making some [keycap/keytopper](https://github.com/openstenoproject/plover/wiki/Supported-Hardware#keytoppers) instead.

By the way, [GitHub issues is not the best place to ask general questions](https://github.com/openstenoproject/plover#getting-help).

**Comment by furry3000** [comment]
> This is possible, but definitely not easy to do. You need a new [system plugin](https://github.com/openstenoproject/plover/wiki/Plugins), and some way to transform the existing dictionary to suit the new system.
> 
> Although in this case, I guess making the thumb span 3 keys is not exactly comfortable. You can try using/making some [keycap/keytopper](https://github.com/openstenoproject/plover/wiki/Supported-Hardware#keytoppers) instead.
> 
> By the way, [GitHub issues is not the best place to ask general questions](https://github.com/openstenoproject/plover#getting-help).

Keycaps didn't work for me. It would just fall off or get in the way. It's easy for me to move the thumb this way: Actually, the thumb is _already_ doing these movements, without a key, since steno uses _double_ keys often. I've been using regular keyboards for decades. I plan to redo the other keys for steno also.  I will check out the getting-help section. Hopefully, it's active.

--Furry3000

Edit: The getting help section actually leads back to here. This is the proper place. This section is open to ideas. It would be interesting to see the other responses.

**Comment by benoit-pierre** [comment]
If the only use case is the one you described (e.g. a key that act as a combo for multiple steno-keys), then you could do that with a custom machine plugin.
